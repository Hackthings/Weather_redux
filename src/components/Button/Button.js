import React, { PureComponent } from 'react';


class Button extends PureComponent {
    render() {
        const { children, ...rest } = this.props;
        return (
            <button {...rest} className="Button">{children}</button>
        )
    }

}

export default Button