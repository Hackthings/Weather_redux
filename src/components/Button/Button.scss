.Button {
    padding: 20px;
    background-color: transparent;
    width: 100%;
    border: 1px solid #ffffff;
    display: block;
    margin: 10px auto;
    font-size: 26px;
    color: #ffffff;
    outline: none;
    cursor: pointer;
    &:hover {
        background-color: rgba(198, 224, 232, 0.25);
        transition: 0.7s;
    }
}