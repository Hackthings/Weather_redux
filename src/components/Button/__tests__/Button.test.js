import React from 'react';
import Button from '../Button';
import { shallow, mount } from 'enzyme';

describe('<Button />', () => {
    it('should redner properly', () => {
        const button = shallow(<Button>test</Button>);
        expect(button).toMatchSnapshot();
    });

    it('should redner children properly', () => {
        const button = shallow(<Button>test</Button>);
        expect(button.text()).toEqual('test');
    })

    it('should handle click', () => {
        const onClick = jest.fn();
        const button = mount(<Button onClick={onClick}>test</Button>);
        button.simulate('click');
        expect(onClick).toHaveBeenCalled();
    })

    it('should handle onmouseover', () => {
        const onMouseOver = jest.fn();
        const button = mount(<Button onMouseOver={onMouseOver}>test</Button>);
        button.simulate('mouseOver');
        expect(onMouseOver).toHaveBeenCalled();
    })


});