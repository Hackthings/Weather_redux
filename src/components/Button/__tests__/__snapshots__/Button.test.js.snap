// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should redner properly 1`] = `
<button
  className="Button"
>
  test
</button>
`;
